firedrake/__init__.py

```diff
@@ -3,11 +3,16 @@
 import sys
 config = firedrake_configuration.get_config()
 if "PETSC_DIR" in os.environ and not config["options"]["honour_petsc_dir"]:
-    raise ImportError("PETSC_DIR is set, but you did not install with --honour-petsc-dir.\n"
-                      "Please unset PETSC_DIR (and PETSC_ARCH) before using Firedrake.")
+    if os.environ["PETSC_DIR"] != os.path.join(os.environ["VIRTUAL_ENV"], "src", "petsc")\
+       or os.environ["PETSC_ARCH"] != "default":
```

Comment:
During testing, because the purest imports Firedrake, which sets PETSC_DIR and PETSC_ARCH on lines 13-15 below, so these environment variables are already set in subprocesses.

Reply:
OK.